### PR TITLE
some refactorings to reduce cyclomatic complexity

### PR DESCRIPTION
**Pull request by jms**
moved some code to functions to reduce the cyclomatic complexity #19

### CODE REVIEW

File: noipy/main.py [hunk 1]
```diff
@@ -34,6 +32,67 @@
...
+    else:
+        user_arg = user_token or utils.read_input(
+            "Type your username: ")
+        pass_arg = password or getpass.getpass("Type your password: ")
+        auth = authinfo.ApiAuth(user_arg, pass_arg)
+
+    return auth
+
+
+def execute_ddns_update(update_ddns, ip, hostname, auth,
```
Comment (pv8) [inline noipy/main.py]: This functions has too many parameters and still has some *Cognitive Complexity*. 
```sh
49-51: Function `execute_ddns_update` has 9 arguments (exceeds 4 allowed). Consider refactoring. [structure]
49-73: Function `execute_ddns_update` has a Cognitive Complexity of 8 (exceeds 5 allowed). Consider refactoring. [structure]
```
Maybe try a different approach here.

File: noipy/main.py [hunk 2]
```diff
@@ -34,6 +32,67 @@
...
+
+    return {
+        'exec_result': exec_result,
+        'response_code': response_code,
+        'response_text': response_text,
+        'process_message': process_message,
+    }
+
+
+def process_generic(url, updater_options,
```
Comment (pv8) [inline noipy/main.py]: Same issues of `execute_ddns_update` function ([line 49](https://github.com/pv8/noipy/pull/25/files#diff-0892748911e7cd8288b84eace1cc35a2R49)):
```sh
77-79: Function `process_generic` has 5 arguments (exceeds 4 allowed). Consider refactoring. [structure]
```

**Comment by pv8** [comment]
@jms, thanks for contributing!
I made some comments on your PR. But, although this PR did reduce code complexity in `execute_update` function, it ends up adding code complexity with the new functions as well. 

Running `codeclimate analyze noipy/`:

- `noipy/main.py` **before** PR #25:
```
== noipy/main.py (2 issues) ==
37-132: Function `execute_update` has a Cognitive Complexity of 40 (exceeds 5 allowed). Consider refactoring. [structure]
37-132: Function `execute_update` has 83 lines of code (exceeds 25 allowed). Consider refactoring. [structure]
```

- `noipy/main.py` **after** PR #25:
```
== noipy/main.py (6 issues) ==
49-51: Function `execute_ddns_update` has 9 arguments (exceeds 4 allowed). Consider refactoring. [structure]
49-73: Function `execute_ddns_update` has a Cognitive Complexity of 8 (exceeds 5 allowed). Consider refactoring. [structure]
77-79: Function `process_generic` has 5 arguments (exceeds 4 allowed). Consider refactoring. [structure]
96-156: Function `execute_update` has a Cognitive Complexity of 15 (exceeds 5 allowed). Consider refactoring. [structure]
96-156: Function `execute_update` has 49 lines of code (exceeds 25 allowed). Consider refactoring. [structure]
96-156: Cyclomatic complexity is too high in function execute_update. (9) [radon]
```

**Comment by jms** [comment]
yes, is no good yet, i'll try a different approach and I'll update this pull request.

**Pull request closed without merging**